### ISSUE TITLE
SAML: changing realname and email address results in new user

### ISSUE THREAD

**Issue by localguru** [RocketChat/Rocket.Chat#6481]
Rocket.Chat Version: 0.52.0

I get a unique and never changing nameid (mapping) from my IDP, like integernumber@domain.tld. This IDP nameid never changes, even if the user's realname or email address or IDM username change. Rocket.Chat as SP creates a new user if those attributes change. Any idea how to avoid that? The idea of an central identity management is having a never changing unique id, even if attributes like real name, email address etc. change and by that unique id one should be mapped always to the same RC user.

Ciao!

**Comment by anicoa** [comment]
So as we have the same problem I digged into the code and found the problematic implementation.

``` 
    let user = Meteor.users.findOne({
      'emails.address': emailRegex
    });

    if (!user) {
```
in steffo_meteor-accounts-saml.js. So at least in our case we got a nameID from IDP and this is what we use (and also which Rocket.Chat uses) for username-creation. This nameID is unique and does never change, but email address of course changes sometimes.

To get a fix here I want to discuss the right solution, i can see two possible approaches

1. Hardcoded: change `'emails.address': emailRegex` to `'username': loginResult.profile.nameID.toLowerCase()`

2. Dynamic: add settings in the admin interface asking for the left side (emails.address) and the right side (loginResult.profile.nameID.toLowerCase()) of the query.

Would be happy to discuss this with everyone here and after that implement it.

Thanks a lot for the great work!
Nico
